test/hardware/test_hardware.py

```
@@ -170,15 +168,67 @@ def test_frost_sign(ser):
+    })
+    assert resp2 is not None, "no response to second commit"
+    assert "error" in resp2, f"second commit should fail but got: {resp2}"
+    msg = resp2["error"]["message"].lower()
+    assert "already" in msg or "duplicate" in msg, f"expected replay error: {resp2}"
+
+    send_receive(ser, {
+        "id": 43, "method": "delete_share",
+        "params": {"group": test_group}
+    })
+
+    print(f"  PASS (replay blocked)")
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)